btcec/ciphering.go

```diff
@@ -14,3 +19,33 @@ import (
 func GenerateSharedSecret(privkey *PrivateKey, pubkey *PublicKey) []byte {
 	return secp.GenerateSharedSecret(privkey, pubkey)
 }
+
+// Encrypt encrypts data for the target public key using AES-128-GCM
+func Encrypt(pubKey *PublicKey, msg []byte) ([]byte, error) {
```

Comment:
Is there a reason why it is AES-128-GCM and not AES-256-CBC as used previously:
https://github.com/Roasbeef/btcd/blob/e8e2167a1af6c20f1f32abb76bd387ee87763bc9/btcec/ciphering.go#L51-L118

Reply:
@Chinwendu20 its actually AES-256-GCM as the shared secret is 32 bytes (256 bits) - would update the comment. I used AES-GCM as it provides inbuilt authentication as compared to AES-CBC.